src/main/scala/mesosphere/mesos/Availability.scala

```diff
@@ -0,0 +1,32 @@
+package mesosphere.mesos
+
+import java.time.Clock
+
+import mesosphere.marathon.RichClock
+import mesosphere.marathon.state.Timestamp
+import org.apache.mesos.Protos.{ DurationInfo, Offer }
+
+import scala.concurrent.duration._
+import scala.language.implicitConversions
+
+object Availability {
+
+  def offerAvailable(offer: Offer, drainingTime: FiniteDuration)(implicit clock: Clock): Boolean = {
+    val now = clock.now()
+    if (offer.hasUnavailability && offer.getUnavailability.hasStart) {
+      val start: Timestamp = offer.getUnavailability.getStart
+      if (now.after(start - drainingTime)) {
```

Comment:
What do you thing about moving that into separate methods to improve readability?